barretenberg/cpp/src/barretenberg/vm2/simulation/execution.cpp

```
@@ -212,6 +212,50 @@ void Execution::op_not(ContextInterface& context, MemoryAddress src_addr, Memory
     }
 }
 
+void Execution::shl(ContextInterface& context, MemoryAddress a_addr, MemoryAddress b_addr, MemoryAddress c_addr)
+{
+    constexpr auto opcode = ExecutionOpCode::SHL;
+    auto& memory = context.get_memory();
+    MemoryValue a = memory.get(a_addr);
+    MemoryValue b = memory.get(b_addr);
+    set_and_validate_inputs(opcode, { a, b });
+
+    get_gas_tracker().consume_gas();
+
+    try {
+        MemoryValue c = alu.shl(a, b);
+        memory.set(c_addr, c);
+        set_output(opcode, c);
+    } catch (const AluException& e) {
+        throw OpcodeExecutionException("SHL Exception: " + std::string(e.what()));
+    } catch (const std::exception& e) {
+        // We have some err not handled by TAG_ERROR:
+        throw OpcodeExecutionException("Alu SHL operation failed");
```

Err im not sure this is correct. The `OpcodeExecutionException` isnt catastrophic so the execution will proceed. This will create an entry in the execution trace without a corresponding alu entry. This will almost certainly break the lookup.

---

If you want to keep this catch, you should just re-throw a runtime error i think. That way it will fall into the `catch (const std::exception& e)` in `Execution::execute`